xdsl/dialects/arith.py

```diff
@@ -110,7 +110,9 @@ class Constant(IRDLOperation):
 
     @overload
     def __init__(
-        self, value: AnyIntegerAttr | FloatAttr[AnyFloat], value_type: None = None
+        self,
+        value: AnyIntegerAttr | FloatAttr[AnyFloat] | Attribute,
```

Comment:
This doesn't seem like a meaningful overload any more, AnyIntegerAttr and FloatAttr[AnyFloat] are all Attribute, so the `|` are superfluous. Maybe what we want here instead is `TypedAttribute`?

Reply:
Also, this really feels like it should be in a separate PR, it is not related to stencils directly, I'm glad I clicked through to see what's in this PR despite the title not saying anything about the standard dialects.

Reply:
I'm getting confused with PRs. I thought we had merged changes to arith and I'd forgotten to make one change, but it's actually still open in #2696, where the `| Attribute` added here is `| DenseIntOrFPElementsAttr`. Rolling back the change here.